### PR TITLE
Add acknowledged checks

### PR DESCRIPTION
**Pull request by cbaines**
| Before | After |
|-----------|---------|
| before | after-2 |

### CODE REVIEW

File: app/styles.css [hunk 1]
```diff
@@ -28,7 +33,7 @@ body {
   color: #fff;
   bottom: 10px;
   display: inline-block;
-  width: calc(100% / 3);
+  width: calc(100% / 4);
```
Comment (ChrisBAshton) [inline app/styles.css]:
```suggestion
width: 25%;
```

File: app/styles.css [hunk 2]
```diff
@@ -3,6 +3,11 @@ body {
   color: #fff;
 }
 
+h2 {
+  margin-top: 0px;
+  margin-bottom: 0px;
+}
```
Comment (ChrisBAshton) [inline app/styles.css:8]:
```css
h2 {
  margin: 0;
}
```

File: app/styles.css [hunk 3]
```diff
@@ -37,15 +42,19 @@ body {
...
   text-align: center;
 }
 
 .blinken-unknown-entries {
-  text-align: right;
+  text-align: center;
+}
+
+.blinken-acknowledged-entries {
+  text-align: center;
```
Comment (ChrisBAshton) [inline app/styles.css]: Can remove all these `text-align: center`s and have a single one:

```css
.blinken-entry {
  text-align: center;
}
```

**Review by ChrisBAshton — COMMENTED** [review]
Styles could be refactored, JS looks good 👍 Thanks for doing this!

**Review by thomasleese — COMMENTED** [review]
I like the idea of tracking acks, however I'm worried that this might make it significantly harder to get it green (which usually has a positive effect of feeling good on 2nd line). Plus, acking an alert is usually a clear message that it's not causing any problems so, in theory, the service is "green" even if there are acked alerts (although I realise it's not always the case in reality).

This might sound silly, but perhaps if the colour for when there are only acked alerts was more green then we could still look at a service and consider it "green" but it would also provide that useful information on the number of acked alerts.

**Comment by bevanloon** [comment]
Implicit in this is that we are changing what it means to be green. Rising to a challenge notwithstanding, is this generally what users (in this case, all the devs working on 2ndline) want?

Any particular reason the new colour was chosen to be what it is (I want to say greyish/bluish but 🤷‍♂)?

**Comment by brucebolt** [comment]
I really like the idea of displaying the number of acknowledged alerts (I don't really have any preference either way about changing the colour).

It's nice to see that there are issues that have been acked, since good practice suggests they should be reviewed regularly to ensure they should remain as such.

File: app/styles.css (hunk 1 above)
Comment (cbaines) [reply app/styles.css]: Thanks, I've changed this now.

File: app/styles.css (hunk 2 above)
Comment (cbaines) [reply app/styles.css:8]: I've made this change now.

File: app/styles.css (hunk 3 above)
Comment (cbaines) [reply app/styles.css]: Yep, I've done this now (in another commit).

**Comment by cbaines** [comment]
> I like the idea of tracking acks, however I'm worried that this might make it significantly harder to get it green (which usually has a positive effect of feeling good on 2nd line).

I guess as I normally use Nagstamon and have it show all the checks, acknowledged and not, I generally have a different picture of the state of things. I'm hopeful that we can rise to the challenge.

> Plus, acking an alert is usually a clear message that it's not causing any problems so, in theory, the service is "green" even if there are acked alerts (although I realise it's not always the case in reality).

If there's a failing check that doesn't relate to a problem, then I say that's a problem with the check configuration. I think acknowledging alerts is still a useful tool, and that's partially why I want more data on it in the display.

> This might sound silly, but perhaps if the colour for when there are only acked alerts was more green then we could still look at a service and consider it "green" but it would also provide that useful information on the number of acked alerts.

My original idea was a "happier" blue colour, but then I make it dark enough to pass the relevant accessibility standards for having white text on top of it, and it ended up quite dark. I've now changed it to a dark green which I'm fine with (see the updated screenshot).

**Comment by cbaines** [comment]
> Implicit in this is that we are changing what it means to be green. Rising to a challenge notwithstanding, is this generally what users (in this case, all the devs working on 2ndline) want?

I haven't heard any objections. I personally hope that the colours of the boxes on the screen aren't taken as a serious indicator as to the health of the platform as a whole (as that's not how our monitoring system works...).

> Any particular reason the new colour was chosen to be what it is (I want to say greyish/bluish but man_shrugging)?

It started off as a lighter blue, then got darker to contrast more with the white text. I've now changed it to a dark green (see the updated screenshots).

**Review by thomasleese — APPROVED** [review]

**Pull request merged**